osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -375,18 +393,34 @@ private void updateScore()
             TotalScore.Value = (long)Math.Round(TotalScoreWithoutMods.Value * scoreMultiplier);
         }
 
+        private readonly Dictionary<HitResult, int> minimumAccuracyResultCounts = new Dictionary<HitResult, int>();
+
         private void updateRank()
         {
             // Once failed, we shouldn't update the rank anymore.
             if (rank.Value == ScoreRank.F)
                 return;
 
             ScoreRank newRank = RankFromScore(Accuracy.Value, ScoreResultCounts);
+            ScoreRank newMaxRank = RankFromScore(MaximumAccuracy.Value, ScoreResultCounts);
+
+            // this arbitrarily only fills HitResult.Miss because it's the only hit result that can affect the state of RankFromScore in the main rulesets (see OsuScoreProcessor).
+            // eventually (whenever it's needed), other miss result types should be correctly filled and this dictionary should be updated regularly next to ScoreResultCounts,
+            // but this is sufficient for now.
+            minimumAccuracyResultCounts[HitResult.Miss] = MaxHits - JudgedHits + ScoreResultCounts.GetValueOrDefault(HitResult.Miss);
```

Comment:
@ppy/team-client requesting review for the addition of this part in particular to resolve https://github.com/ppy/osu/pull/28614#discussion_r1670392366

Reply:
Too hacky to live imo. The remaining results should be populated so that this is in a _sane_ state.

Also I'm not even convinced this is right - what if some of the remaining judgements are `IgnoreMiss`, for example?

Reply:
Partly addressed in https://github.com/ppy/osu/pull/28614/commits/471d7f2aedf97800c7b2bd1367855524f9bcd7c1, see https://github.com/ppy/osu/pull/28614#issuecomment-2436363284.